src/annotator/hypothesis-injector.js

```diff
@@ -95,24 +50,57 @@ function hasHypothesis(iframe) {
 }
 
 /**
- * Inject the client's boot script into the iframe. The iframe must be from the
- * same origin as the current window.
+ * Inject Hypothesis client into a frame.
  *
- * @param {HTMLIFrameElement} iframe
- * @param {string} scriptSrc
- * @param {Record<string, any>} config
+ * IMPORTANT: This method requires that the iframe is "accessible"
```

Comment:
```suggestion
 * IMPORTANT: This method requires that the iframe is same-origin
```

The specific test used to determine whether one document can reach into another is whether they are [same-origin](https://html.spec.whatwg.org/multipage/browsers.html#concept-bcc-content-document).